core/domain/html_cleaner.py

```diff
@@ -782,11 +783,202 @@ def add_caption_attr_to_image(html_string):
         str. Updated HTML string with the caption attribute for all
             oppia-noninteractive-image tags.
     """
-    soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
 
     for image in soup.findAll(name='oppia-noninteractive-image'):
         attrs = image.attrs
         if 'caption-with-value' not in attrs:
             image['caption-with-value'] = escape_html(json.dumps(''))
 
     return unicode(soup)
+
+
+def validate_customization_args(html_list):
+    """Validates customization arguments of Rich Text Components in a list of
+    html string.
+
+    Args:
+        html_list: list(str). List of html strings to be validated.
+
+    Returns:
+        dict: Dictionary of all the invalid customisation args where
+            key is a Rich Text Component and value is the invalid html string.
+    """
+    # Dictionary to hold html strings in which customization arguments
+    # are invalid.
+    err_dict = {}
+
+    for html_string in html_list:
+        soup = bs4.BeautifulSoup(
+            html_string.encode(encoding='utf-8'), 'html.parser')
+        _validate_customization_args_in_soup(soup, err_dict, html_string)
+
+        for collapsible in soup.findAll(
+                name='oppia-noninteractive-collapsible'):
+            # This is wrapped in try catch block since any invalid
+            # content such as int will not be convertible in Beautiful
+            # Soup format.
+            try:
+                content_html = json.loads(
+                    unescape_html(collapsible['content-with-value']))
+                soup_for_collapsible = bs4.BeautifulSoup(
+                    content_html, 'html.parser')
+                _validate_customization_args_in_soup(
```

Comment:
Generally we keep try/catch as small as possible. Can this go outside the try/catch?

How about the json.loads on line 823?

Reply:
I realized this cannot be done since if we try to load any customization arg which is missing that will produce an error too, so this complete block should be in try catch blocks. For example, if `tabs_content-with-value` is missing, trying to obtain it from attrs dict will raise a KeyError.

Reply:
I have updated the comment to reflect the same.